tests/components/datadog/test_init.py

```diff
@@ -10,59 +9,47 @@
+        assert not await async_setup_component(
+            hass, datadog.DOMAIN, {datadog.DOMAIN: {"host1": "host1"}}
+        )
 
-    def setUp(self):  # pylint: disable=invalid-name
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.addCleanup(self.tear_down_cleanup)
 
-    def tear_down_cleanup(self):
-        """Stop everything that was started."""
-        self.hass.stop()
+async def test_datadog_setup_full(hass):
+    """Test setup with all data."""
+    config = {datadog.DOMAIN: {"host": "host", "port": 123, "rate": 1, "prefix": "foo"}}
+    hass.bus.listen = MagicMock()
 
-    def test_invalid_config(self):
-        """Test invalid configuration."""
-        with assert_setup_component(0):
-            assert not setup_component(
-                self.hass, datadog.DOMAIN, {datadog.DOMAIN: {"host1": "host1"}}
-            )
+    with patch("homeassistant.components.datadog.initialize") as mock_init, patch(
+        "homeassistant.components.datadog.statsd"
+    ):
+        assert await async_setup_component(hass, datadog.DOMAIN, config)
 
-    @mock.patch("homeassistant.components.datadog.statsd")
-    @mock.patch("homeassistant.components.datadog.initialize")
-    def test_datadog_setup_full(self, mock_connection, mock_client):
-        """Test setup with all data."""
-        self.hass.bus.listen = mock.MagicMock()
+        assert mock_init.call_count == 1
+        assert mock_init.call_args == mock.call(statsd_host="host", statsd_port=123)
 
-        assert setup_component(
-            self.hass,
-            datadog.DOMAIN,
-            {datadog.DOMAIN: {"host": "host", "port": 123, "rate": 1, "prefix": "foo"}},
-        )
+    assert hass.bus.listen.called
+    assert EVENT_LOGBOOK_ENTRY == hass.bus.listen.call_args_list[0][0][0]
+    assert EVENT_STATE_CHANGED == hass.bus.listen.call_args_list[1][0][0]
 
-        assert mock_connection.call_count == 1
-        assert mock_connection.call_args == mock.call(
-            statsd_host="host", statsd_port=123
-        )
 
-        assert self.hass.bus.listen.called
-        assert EVENT_LOGBOOK_ENTRY == self.hass.bus.listen.call_args_list[0][0][0]
-        assert EVENT_STATE_CHANGED == self.hass.bus.listen.call_args_list[1][0][0]
+async def test_datadog_setup_defaults(hass):
+    """Test setup with defaults."""
+    hass.bus.listen = mock.MagicMock()
 
-    @mock.patch("homeassistant.components.datadog.statsd")
-    @mock.patch("homeassistant.components.datadog.initialize")
-    def test_datadog_setup_defaults(self, mock_connection, mock_client):
-        """Test setup with defaults."""
-        self.hass.bus.listen = mock.MagicMock()
-
-        assert setup_component(
-            self.hass,
+    with patch("homeassistant.components.datadog.initialize") as mock_init, patch(
+        "homeassistant.components.datadog.statsd"
+    ):
+        assert await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)